core/src/main/java/org/infinispan/protostream/impl/ProtoStreamWriterImpl.java

```diff
@@ -95,9 +92,6 @@ public void writeInt(String fieldName, Integer value) throws IOException {
       final FieldDescriptor fd = messageContext.getFieldByName(fieldName);
       checkFieldWrite(fd);
       if (value == null) {
-         if (fd.isRequired()) {
-            throw new IllegalArgumentException("A required field cannot be null : " + fd.getFullName());
```

Comment:
Do we still throw an exception elsewhere if the user configures proto2 and uses a null value? I wouldn't expect this behaviour to change when `required=true`.

Reply:
I've restored the functionality via an alternate way

Reply:
This needs to be applied to all of the `write*` methods.